src/3d/qgs3dmapsettings.cpp

```diff
@@ -293,6 +298,10 @@ QDomElement Qgs3DMapSettings::writeXml( QDomDocument &doc, const QgsReadWriteCon
   QDomElement elemCamera = doc.createElement( QStringLiteral( "camera" ) );
   elemCamera.setAttribute( QStringLiteral( "field-of-view" ), mFieldOfView );
   elemCamera.setAttribute( QStringLiteral( "projection-type" ), static_cast< int >( mProjectionType ) );
+  if ( mCameraNavigationMode == QgsCameraController::BasicNavigation )
```

Comment:
use a switch statement here (that's general good practice for enum based conditionals)